app/src/org/commcare/fragments/connectId/ConnectIdBiometricConfigFragment.java

```diff
@@ -100,10 +100,12 @@ public void onAuthenticationError(int errorCode,
                     }
                 }
 
-                Logger.exception("Exhausted biometrics", new Exception(
-                        "Fingerprint error and PIN isn't configured/allowed: " + allowPassword));
+                String whyNoPin = allowPassword ? "configured" : "allowed";
+                Logger.exception("Exhausted biometrics", new Exception(String.format(Locale.getDefault(),
+                        "Fingerprint error and PIN isn't %s: %s (%d)", whyNoPin, errString, errorCode)));
 
-                Toast.makeText(context, R.string.connect_verify_configuration_failed, Toast.LENGTH_SHORT).show();
+                String message = context.getString(R.string.connect_verify_configuration_failed, errString);
+                Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
```

Comment:
we should use `LENGTH_LONG` to give user enough time to read as it might be a long string with the `errString`